evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -59,25 +60,24 @@ def validate(self, X, y):
             raise ValueError("y cannot be None")
 
         y = _convert_to_woodwork_structure(y)
-        y = _convert_woodwork_types_wrapper(y.to_series())
-
-        null_rows = y.isnull()
+        is_supported_type = y.logical_type in numeric_and_boolean_ww + [ww.logical_types.Categorical]
+        if not is_supported_type:
+            messages["errors"].append(DataCheckError(message="Target is unsupported {} type. Valid Woodwork logical types include: {}"
```

Comment:
@angela97lin so we're doing one type-check here for all the problem types? Is there any value to having this vary per problem type, i.e. disallow `Categorical` for regression problems?
```python
supported_types = numeric_and_boolean_ww
if is_regression(problem_type):
    supported_types += [ww.logical_types.Categorical]
if not y.logical_type in supported_types:
    messages["errors"].append(DataCheckError(message="Target is unsupported type {}. Valid Woodwork logical types include: {}".format(y.logical_type, ", ".join([t.type_string for t in supported_types]))
```

Similarly we could disallow binary for multiclass problems

Reply:
@dsherry I wanted to use this PR to strictly focus on converting our logic to the WW equivalent and not change too much behavior. You raise a great point though, and I previously already filed https://github.com/alteryx/evalml/issues/1548 to track this :)

Reply:
Amazing :)